src/net.cpp

```diff
@@ -1543,30 +1552,55 @@ void CConnman::ThreadDNSAddressSeed()
     Shuffle(seeds.begin(), seeds.end(), rng);
     int seeds_right_now = 0; // Number of seeds left before testing if we have enough connections
     int found = 0;
+    const std::chrono::seconds seeds_wait_time = (addrman.size() >= 1000 ? DNSSEEDS_DELAY_MANY_PEERS : DNSSEEDS_DELAY_FEW_PEERS);
```

Comment:
What the logic for picking 1000 ? If we assume uniform address propagation, what's the ratio of reliable public nodes on others in addrman and what the probability of not connecting successfully to at least 3 outbound peers ? Based on that, I would say intuitively we may scale down number a bit to increase number of nodes who delay DNS query.

Reply:
1000 is the maximum number of entries the `addr` message can include, so seems a good threshold to indicate you've managed to pick up some live data from the p2p network about potential peers, rather than just some results from DNS lookups.

Reply:
Okay, make sense given we send `GETADDR` at `VERSION` reception which means if you have at only one reliable outbound connection you should be able to bootstrap your addrman without seed assistance.